src/neo4j_graphrag/experimental/pipeline/pipeline.py

```diff
@@ -412,6 +416,59 @@ def validate_parameter_mapping_for_task(self, task: TaskPipelineNode) -> bool:
     async def get_final_results(self, run_id: str) -> dict[str, Any]:
         return await self.final_results.get(run_id)  # type: ignore[no-any-return]
 
+    async def stream(self, data: dict[str, Any]) -> AsyncGenerator[Event, None]:
+        """Run the pipeline and stream events for task progress.
+
+        Args:
+            data: Input data for the pipeline components
+
+        Yields:
+            Event: Pipeline and task events including start, progress, and completion
+        """
+        # Create queue for events
+        event_queue: asyncio.Queue[Event] = asyncio.Queue()
+
+        async def event_stream(event: Event) -> None:
+            # Put event in queue for streaming
+            await event_queue.put(event)
+
+        # Add event streaming callback
+        self.callbacks.append(event_stream)
+
+        event_queue_getter_task = None
+        try:
+            # Start pipeline execution in background task
+            run_task = asyncio.create_task(self.run(data))
+
+            # loop until the run task is done, and we do not have
+            # any more pending tasks in queue
+            is_run_task_running = True
+            is_queue_empty = False
+            while is_run_task_running or not is_queue_empty:
+                # Wait for next event or pipeline completion
+                event_queue_getter_task = asyncio.create_task(event_queue.get())
+                done, pending = await asyncio.wait(
+                    [run_task, event_queue_getter_task],
+                    return_when=asyncio.FIRST_COMPLETED,
+                )
+
+                is_run_task_running = run_task not in done
+                is_queue_empty = event_queue.empty()
+
+                for event_future in done:
+                    if event_future == run_task:
+                        continue
+                    yield event_future.result()  # type: ignore
```

Comment:
What errors do you get without `type: ignore`?

Reply:
>  Incompatible types in "yield" (actual type "BaseModel", expected type "Event")

I have no clue where the `BaseModel` comes from here, the queue is defined as `Queue[Event]` so it should always return an `Event` (and this is what I observed in my tests)